runtime/src/snapshot_utils.rs

```diff
@@ -1321,8 +1334,31 @@ pub fn purge_old_snapshot_archives<P>(
             }
         });
 
+    if !incremental_snapshot_archives_with_different_base_slot.is_empty() {
+        info!(
+            "Purging {} incremental snapshot archives with a different base slot than the highest full snapshot slot",
+            incremental_snapshot_archives_with_different_base_slot.len()
+        );
+    } else {
+        info!("There are no incremental snapshot archives with a different base slot than the highest full snapshot slot");
+    }
```

Comment:
IMO, no need to log if we're not doing anything.

```suggestion
    if !incremental_snapshot_archives_with_different_base_slot.is_empty() {
        trace!(
            "Purging {} incremental snapshot archives with a different base slot than the highest full snapshot slot",
            incremental_snapshot_archives_with_different_base_slot.len()
        );
    }
```